### ISSUE TITLE
Stripe not getting email address [$100]

### ISSUE THREAD

**Issue by cassidyjames** [elementary/mvp#340]
Stripe will automatically send an email receipt upon successful payment. However, it appears Stripe isn't getting an email address from our site. When viewing transactions in the Stripe dashboard, no email addresses are listed and no receipts have been sent.

**Comment by cassidyjames** [comment]
`receipt_email: null` on the charge.succeeded event details.

**Comment by lewisgoddard** [comment]
I don't get this with test keys. 
```
card:
    ...
    brand: "Visa"
    country: "US"
    cvc_check: "pass"
    exp_month: 12
    exp_year: 2015
    funding: "credit"
    id: "card_5gLvMd679EDf8E"
    last4: "4242"
    name: "goddard.lewis@gmail.com"
....
client_ip: ...
created: 1423656259
email: "goddard.lewis@gmail.com"
id: ...
livemode: false
object: "token"
type: "card"
used: false
verification_allowed: true
```

**Comment by cassidyjames** [comment]
@lewisgoddard what's the progress of this? I consider this critical; we need to be sending receipts as soon as possible.

**Comment by lewisgoddard** [comment]
On my branch, I've got nothing that I can see to fix this through Stripe. I don't know their API well enough to see where we're going wrong, short of just raising the issue with their support.

As for doing our own custom ones, if someone gives me a design and Mandrill or Mailgun account details, I could have it done by the end of the week, (faster with Mailgun because I have a wrapper). https://github.com/elementary/mvp/issues/34#issuecomment-69489692

**Comment by KeitIG** [comment]
Do you want I take a look tonight ?

**Comment by lewisgoddard** [comment]
I've effectively left #381 because I don't know how to proceed with it, you can take over if you wish, or start afresh. I am not currently working on this bug.

**Comment by funnylookinhat** [comment]
Hey guys -
I'm fairly certain  you only need to add one line to this section:
https://github.com/elementary/mvp/blob/master/backend/payment.php#L14-L19

```
$charge = Stripe_Charge::create(array(
    'amount' => $amount,
    'currency' => 'usd',
    'card' => $token,
    'description' => 'elementary OS download',
    'receipt_email' => "user email address", // Set this value appropriately
));
```

I didn't see where you're storing customer email addresses at first glance, but you're obviously more familiar with the codebase than I am.  Putting it in there should start sending receipts.

Here's the API Reference in case you want to add anything more to the charge:
https://stripe.com/docs/api#create_charge

Cheers.

**Comment by funnylookinhat** [comment]
Ah - I take that back.  You need to send it from here:
https://github.com/elementary/mvp/blob/6607537dd3be4443f4b858c08abca1cec3cfeee7/scripts/homepage.js#L78

Just add the user's email address to your form or whatnot and then reference it as $_POST['whatevz'] in your Stripe_Charge::create call.

**Comment by lewisgoddard** [comment]
I've got a slightly different but testable branch up at http://beta.elementary.io/branch/stripe-receipts-3
Pull request '456

**Comment by funnylookinhat** [comment]
@lewisgoddard Your changes look pretty similar to mine - but a bit more comprehensive...  I guess I wasn't comprehending what the exact problem is.  Simply getting emails added to receipts should be as easy as those two lines though ( but your code is a bit better, I forgot encodeURIComponent on the email address ).

**Comment by lewisgoddard** [comment]
I think you had this issue down, but my original branch was fixing some other problems, and ended up messy and broken. I'm just covering a few extra edge cases is all.

**Comment by lewisgoddard** [comment]
Fixed in 50ffe677d98fb42ad110ac572cf9f83fd38acdd8

**Comment by cassidyjames** [comment]
Thanks for getting this wrapped up @lewisgoddard, and thanks for jumping in and taking a look, @funnylookinhat. I'll poke you again if we run into any other issues. :)
